packages/fields/src/actions/set-as-homepage.tsx

```diff
@@ -0,0 +1,364 @@
+/**
+ * WordPress dependencies
+ */
+import { select, useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import type { Settings, Page } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	TextControl,
+	SelectControl,
+	Spinner,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+} from '@wordpress/components';
+import type { Action, ActionModal } from '@wordpress/dataviews';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from './utils';
+import type { CoreDataError, PostWithPermissions } from '../types';
+
+const PAGE_POST_TYPE = 'page';
+
+const useSiteSettings = () =>
```

Comment:
Probably no need to have extra hook for this. Why not just include in the component that uses it?

Reply:
Ah thanks, I think removing this extra hook works well! Done here: https://github.com/WordPress/gutenberg/pull/65426/commits/40568dcf820692639d16ef6f9a2a6dd944dbb1e1.